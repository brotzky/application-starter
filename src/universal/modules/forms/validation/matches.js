import validator from 'validator';

export default function(match, err = 'is not valid') {
  return function(value) {
    return validator.matches(value, match)
      ? undefined
      : {
          defaultMessage: `${err.defaultMessage}`,
          id: err.id || 'form.errors.matches',
          values: { err: err },
        };
  };
}
