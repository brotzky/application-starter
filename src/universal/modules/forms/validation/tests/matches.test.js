import matches from '../matches';

const matchesHello = matches('hello');

describe('[Form Validation] matches', () => {
  describe('matches()', () => {
    it('should return a function', () => {
      expect(matches('hello')).toMatchSnapshot();
    });

    it('should return no errors if the string matches string originally defined', () => {
      expect(matchesHello('hello')).toMatchSnapshot();
    });

    it('should return an error object when the string does not match the specific one', () => {
      expect(matchesHello('Hello')).toMatchSnapshot();
    });
  });
});
