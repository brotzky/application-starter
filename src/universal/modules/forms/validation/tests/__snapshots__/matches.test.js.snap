// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] matches matches() should return a function 1`] = `[Function]`;

exports[`[Form Validation] matches matches() should return an error object when the string does not match the specific one 1`] = `
Object {
  "defaultMessage": "undefined",
  "id": "form.errors.matches",
  "values": Object {
    "err": "is not valid",
  },
}
`;

exports[`[Form Validation] matches matches() should return no errors if the string matches string originally defined 1`] = `undefined`;
